modules/gdscript/gdscript.cpp

```diff
@@ -2080,20 +2080,26 @@ void GDScriptInstance::notification(int p_notification, bool p_reversed) {
 	//notification is not virtual, it gets called at ALL levels just like in C.
 	Variant value = p_notification;
 	const Variant *args[1] = { &value };
+	const StringName &notification_str = GDScriptLanguage::get_singleton()->strings._notification;
 
-	List<GDScript *> pl;
-	GDScript *sptr = script.ptr();
-	while (sptr) {
-		if (p_reversed) {
-			pl.push_back(sptr);
+	static LocalVector<GDScript *> script_stack;
+	uint32_t script_count = 0;
+	for (GDScript *sptr = script.ptr(); sptr; sptr = sptr->_base, ++script_count) {
+		if (unlikely(script_stack.size() == script_count)) {
+			script_stack.push_back(sptr);
 		} else {
-			pl.push_front(sptr);
+			script_stack[script_count] = sptr;
```

Comment:
Without the `static` you also don't need the if.

```suggestion
		script_stack.push_back(sptr);
```